packages/webpack-dynamic-bundle-plugin/src/batch-dynamic-plugin.ts

```diff
@@ -0,0 +1,193 @@
+                  Template.indent(
+                    `script.crossOrigin = ${JSON.stringify(crossOriginLoading)};`
+                  ),
+                  "}"
+                ])
+                : "",
+              "// create error before stack unwound to get useful stacktrace later",
+              "var error = new Error();",
+              "onScriptComplete = function (event) {",
+              Template.indent([
+                "// avoid mem leaks in IE.",
+                "script.onerror = script.onload = null;",
+                "clearTimeout(timeout);",
+                // DIFF: We have multiple chunks in the batch, so we have to iterate.
+                // Originally (in webpack 5), the `loadingEnded` function replaces
+                // `chunk.done()`. We capture those callbacks above.
+                "for (var chunk of batch) {",
+                Template.indent([
+                  "var reportError = chunk.done();",
+                  "if(reportError) {",
+                  Template.indent([
+                    "var errorType = event && (event.type === 'load' ? 'missing' : event.type);",
+                    "var realSrc = event && event.target && event.target.src;",
+                    "error.message = 'Loading chunk ' + chunk.id + ' failed.\\n(' + errorType + ': ' + realSrc + ')';",
+                    "error.name = 'ChunkLoadError';",
+                    "error.type = errorType;",
+                    "error.request = realSrc;",
+                    "reportError(error);"
+                  ]),
+                  "}",
+                ]),
+                "}"
+              ]),
+              "};",
+              "var timeout = setTimeout(function(){",
+              Template.indent([
+                "onScriptComplete({ type: 'timeout', target: script });"
+              ]),
+              `}, ${chunkLoadTimeout});`,
+              "script.onerror = script.onload = onScriptComplete;"
+            ]),
+            "});"
+          ]),
+          `} else {`,
+          Template.indent([
+            // DIFF: The surrounding code assumes that this code always generates
+            // a local `script` variable that can be appended to the DOM. So even
+            // when we're adding to a batch, we need to generate some kind of DOM
+            // node. So we're adding comments. Also helps with debugging!
+            "var onScriptComplete;",
+            `${activeBatch}.push(batchEntry);`,
+            "script = document.createComment(' Chunk ' + chunkId + ' queued ');",
+          ]),
+          `}`,
+        ]);
+      });
+    });
+  }
+}
+
+/**
+ * Simulate the chunk loaded callback in webpack 4. Could be dropped if we don't
+ * care about working with webpack 4. This function is stringified and will run
+ * in the browser. `installedChunks` comes from the scope it gets inserted into.
+ */
+function loadingEnded4() {
+  var returnValue = null;
+  // @ts-ignore
```

Comment:
Why is this needed? Is that because webpack4 is not well-typed?

Reply:
The problem is that this function is serialized and inserted into a different scope. Added a local comment to that effect, more details are in the function-level comment.